blivet/devices/storage.py

```diff
@@ -243,7 +243,7 @@ def _setTargetSize(self, newsize):
         if not isinstance(newsize, Size):
             raise ValueError("new size must of type Size")
 
-        if self.maxSize and newsize > self.maxSize:
+        if self.maxSize and newsize and newsize > self.maxSize:
```

Comment:
Hmm, okay. Does the newly-empty thin-pool get removed eventually? If so, from where?